packages/flutter_markdown/lib/src/markdown_raw.dart

```diff
@@ -0,0 +1,400 @@
+    this.markdownStyle,
+    this.syntaxHighlighter
+  });
+
+  /// Markdown styled text
+  final String data;
+
+  /// Style used for rendering the markdown
+  final MarkdownStyleRaw markdownStyle;
+
+  /// The syntax highlighter used to color text in code blocks
+  final SyntaxHighlighter syntaxHighlighter;
+
+  _MarkdownState createState() => new _MarkdownState();
+
+  MarkdownStyleRaw createDefaultStyle(BuildContext context) {
+    return null;
+  }
+}
+
+class _MarkdownState extends State<MarkdownBodyRaw> {
+
+  void initState() {
+    super.initState();
+
+    MarkdownStyleRaw markdownStyle = config.markdownStyle ?? config.createDefaultStyle(context);
+    SyntaxHighlighter syntaxHighlighter = config.syntaxHighlighter ?? new _DefaultSyntaxHighlighter(markdownStyle.code);
+
+    _cachedBlocks = _blocksFromMarkup(config.data, markdownStyle, syntaxHighlighter);
+  }
+
+  List<_Block> _cachedBlocks;
+
+  Widget build(BuildContext context) {
+    List<Widget> blocks = <Widget>[];
+    for (_Block block in _cachedBlocks) {
+      blocks.add(block.build(context));
+    }
+
+    return new Column(
+      alignItems: FlexAlignItems.stretch,
+      children: blocks
+    );
+  }
+}
+
+List<_Block> _blocksFromMarkup(String data, MarkdownStyleRaw markdownStyle, SyntaxHighlighter syntaxHighlighter) {
+  List<String> lines = data.replaceAll('\r\n', '\n').split('\n');
+  md.Document document = new md.Document();
+
+  _Renderer renderer = new _Renderer();
+  return renderer.render(document.parseLines(lines), markdownStyle, syntaxHighlighter);
+}
+
+class _Renderer implements md.NodeVisitor {
+  List<_Block> render(List<md.Node> nodes, MarkdownStyleRaw markdownStyle, SyntaxHighlighter syntaxHighlighter) {
+    assert(markdownStyle != null);
+
+    _blocks = <_Block>[];
+    _listIndents = <String>[];
+    _markdownStyle = markdownStyle;
+    _syntaxHighlighter = syntaxHighlighter;
+
+    for (final md.Node node in nodes) {
+      node.accept(this);
+    }
+
+    return _blocks;
+  }
+
+  List<_Block> _blocks;
+  List<String> _listIndents;
+  MarkdownStyleRaw _markdownStyle;
+  SyntaxHighlighter _syntaxHighlighter;
+
+  void visitText(md.Text text) {
+    List<dynamic> top = _currentBlock.stack.last;
+
+    if (_currentBlock.tag == 'pre')
+      top.add(_syntaxHighlighter.format(text.text));
+    else
+      top.add(text.text);
+  }
+
+  bool visitElementBefore(md.Element element) {
+    if (_isListTag(element.tag))
+      _listIndents.add(element.tag);
+
+    if (_isBlockTag(element.tag)) {
+      List<_Block> blockList;
+      if (_currentBlock == null)
+        blockList = _blocks;
+      else
+        blockList = _currentBlock.subBlocks;
+
+      _Block newBlock = new _Block(element.tag, element.attributes, _markdownStyle, new List<String>.from(_listIndents), blockList.length);
+      blockList.add(newBlock);
+    } else {
+      TextStyle style = _markdownStyle.styles[element.tag] ?? new TextStyle();
+      List<dynamic> styleElement = <dynamic>[style];
```

Comment:
It looks like you're still using the old-style StyledText data structures.  I thought we discussed changing this code to be well-typed.